amundsen-kube-helm/README.md

```diff
@@ -102,9 +99,14 @@ config:
     pagecache_size: 2Gi
 ```
 
-With this values file, you can then setup amundsen with these commands:
+With this values file, you can then install Amundsen using Helm 2 with:
+``` shell
+helm install ./templates/helm --values impl/helm/dev/values.yaml
 ```
-helm install templates/helm/amundsen --values impl/helm/dev/values.yaml
+
+For Helm 3 it's now mandatory to specify a [chart reference name](https://helm.sh/docs/intro/using_helm/#helm-install-installing-a-package) e.g. `my-amundsen`: 
+``` shell
+helm install my-amundsen ./templates/helm --values impl/helm/dev/values.yaml
```

Comment:
This doesn't work for me either. What is passed into the `--values` flag? My own custom values right?

Reply:
I got this error when I ran the command without the --values part:
```
helm install dsci-amundsen ./templates/helm
Error: found in Chart.yaml, but missing in charts/ directory: elasticsearch
vvv

Reply:
Oh yes, you're right about the missing elasticsearch. See if running this doesn't fix it:
``` shell
helm dependency update 
```

Reply:
Regarding the `--values` flag. Yes it's to override with your own values. I just went back in git history and found an old snippet https://github.com/lyft/amundsen/tree/7c58452dc4917390602846afcf542cd285a9ae4d/amundsen-kube-helm#how-do-i-use-this that used to sit before the **"With this values file, you can then setup..."** 

Maybe something like that in an updated form should be added again?

Reply:
insisde the amundsen-kube-helm folder: 
```
 helm dependency update
Error: validation: chart.metadata is required
```

Reply:
step by step, what did you do the install this in your k8 cluster?

Reply:
Whoops sorry, I was actually running my helm dependency command from within the chart folder (`./templates/helm`) so it apparently picked up the CHART parameter with `.` as default. So inside the amundsen-kube-helm folder, try:
``` shell
helm dependency update ./templates/helm
```

Reply:
So everything from the top (as much as I can, in WSL, on a PC, with my clone directory being `C:\Hub\amundsen_220tlp_only`) - with me:

``` bash
$ pwd
/mnt/c/Hub/amundsen_220tlp_only/amundsen-kube-helm
$ helm version
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
$ helm dependency update ./templates/helm/
Saving 1 charts
Downloading elasticsearch from repo https://kubernetes-charts.storage.googleapis.com/
Deleting outdated charts
$ helm install my-amundsen ./templates/helm/ --debug --dry-run
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /mnt/c/Hub/amundsen_220tlp_only/amundsen-kube-helm

NAME: my-amundsen
LAST DEPLOYED: Tue Apr 14 00:07:21 2020
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
LONG_RANDOM_STRING: 1234
affinity: {}
dnsZone: teamname.company.com
dockerhubImagePath: amundsendev
..... 
```

Reply:
Hi @jornh 

I am getting this error when I run the above:
```
helm install my-amundsen ./templates/helm/ --debug --dry-run
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /home/joel/AISG/UDP/amundsen/amundsen-kube-helm/templates/helm

Error: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1", unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta1"]
helm.go:75: [debug] [unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1", unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta1"]
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).Run
        /home/circleci/helm.sh/helm/pkg/action/install.go:247
main.runInstall
        /home/circleci/helm.sh/helm/cmd/helm/install.go:223
main.newInstallCmd.func1
        /home/circleci/helm.sh/helm/cmd/helm/install.go:117
github.com/spf13/cobra.(*Command).execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
        /home/circleci/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1357
```

Reply:
Huh! That error isn’t giving very clear guidance ... I wonder where the `”apps/v1beta1”` comes from? Something you added in your environment at some point - from the Amundsen/elastic charts - or from helm itself? What version of helm are you on? Did it fetch the elasticsearch chart without problems?

I experienced some similarly confusing, but different, error message yesterday. It turned out I had some Amundsen Kubernetes objects lingering even after doing a `helm uninstall` of Amundsen. I found `kubectl get all --all-namespaces` to be useful there.

Maybe worth trying to do everything in a different namespace than “default” if it’s polluted in some way... (or to rule that out at least) you’d have to look up the helm syntax for it. I’m in no way a helm superuser either. I just happened to get it spun up ... unfortunately with some missteps etc. and learning along the way it’s easy to miss documenting a step.

Reply:
i think it has to do with kubernetes 1.18 under the release notes:

> the following deprecated APIs can no longer be served:
> All resources under apps/v1beta1 and apps/v1beta2 - use apps/v1 instead

Reply:
@Caisho OK, well googled! Sounds reasonable.

 It sounds like a Kubernetes 1.18 compatibility issue outside the original scope of this PR. Please open a separate issue for tracking that if you think it’s something related to Amundsen.

Reply:
@Caisho Maybe [this](https://github.com/lyft/amundsen/pull/483 ) PR fixed your issue.